modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-pricing-impl/src/main/java/com/liferay/headless/commerce/admin/pricing/internal/resource/v2_0/PriceEntryResourceImpl.java

```diff
@@ -319,6 +326,16 @@ private Map<String, Map<String, String>> _getActions(
 		).build();
 	}
 
+	private BigDecimal _getBigDecimal(Object value, BigDecimal defaultValue) {
+		Number number = GetterUtil.get(value, defaultValue);
+
+		if (number == null) {
+			return null;
+		}
+
+		return BigDecimal.valueOf(number.longValue());
```

Comment:
this will lead you in losing decimal precision, use num.doubleValue instead or get rid of this method and cast directly to BigDecimal where you use this method

Reply:
Removed the method and used the cast to BigDecimal